src/Adaptivity/constraints.jl

```diff
@@ -0,0 +1,42 @@
+"""
+    This constraint can be passed to the constraint handler when working with non-conforming meshes to
+    add the affine constraints required to make the associated interpolation conforming.
+
+    For a full example visit the AMR tutorial.
+"""
+struct ConformityConstraint
+    field_name::Symbol
+end
+
+function Ferrite.add!(ch::ConstraintHandler{<:DofHandler{<:Any,<:Grid}}, cc::ConformityConstraint)
+    @warn "Trying to add conformity constraint to $(cc.field_name) on a conforming grid. Skipping."
+end
+
+function Ferrite.add!(ch::ConstraintHandler{<:DofHandler{<:Any,<:NonConformingGrid}}, cc::ConformityConstraint)
+    @assert length(ch.dh.field_names) == 1 "Multiple fields not supported yet."
```

Comment:
For a dofhandler with two scalar fields, the present function can apply the conformity constraint for hanging nodes. So I think that the assert can be replaced with a warning. 
```suggestion
    @assert length(ch.dh.field_names) == 1 "Multiple fields not supported yet."
```